app/javascript/shared/utils.ts

```diff
@@ -102,18 +103,126 @@ class ResponseError extends Error {
   }
 }
 
-export function getJSON(url: string, data: unknown, method = 'GET') {
-  const { query, ...options } = fetchOptions(data, method);
+const FETCH_TIMEOUT = 30 * 1000; // 30 sec
+
+// Perform a GET HTTP request using `fetch` API,
```

Comment:
Oups, c'est moi qui ait écrit un commentaire trompeur :

```suggestion
// Perform an HTTP request using `fetch` API,
```